### ISSUE TITLE
ActiveSupport::Notifications instrumentation

### ISSUE THREAD

**Issue by nicolas-brousse** [github/view_component#427]
## Feature request
Have `ActiveSupport::Notifications` instrumentation.

### Motivation

It may be nice to have some information about time used to render the component. Like `ActionView` does in developer server logs.

Examples:
- simple component rendering `Rendered ButtonComponent (Duration: 1.2ms | Allocations: 123)`
- collection rendering `Rendered collection of ButtonComponent [10 times] (Duration: 1.2ms | Allocations: 123)`

```
Started GET "/posts/aa147930-7749-4d52-a44b-29548e093447/edit" for 127.0.0.1 at 2020-07-29 01:09:19 +0200
Processing by PostsController#edit as HTML
  Parameters: {"id"=>"aa147930-7749-4d52-a44b-29548e093447"}
  Post Load (2.0ms)  SELECT "posts".* FROM "posts" WHERE "posts"."id" = $1 LIMIT $2  [["id", "aa147930-7749-4d52-a44b-29548e093447"], ["LIMIT", 1]]
  ↳ app/controllers/posts_controller.rb:51:in `set_post'
  Rendering posts/edit.html.slim within layouts/application
  Rendered ButtonComponent (Duration: 1.2ms | Allocations: 123)
  Rendered posts/_form.html.slim (Duration: 6.3ms | Allocations: 3331)
  Rendered posts/edit.html.slim within layouts/application (Duration: 9.4ms | Allocations: 5561)
Completed 200 OK in 46ms (Views: 26.1ms | ActiveRecord: 2.3ms | Allocations: 19197)
```

**Comment by joelhawksley** [comment]
@nicolas-brousse that would be nice! Do you know if ViewComponent render times are currently included in the existing ActionView stats? We are rendering through ActionView after all.

**Comment by nicolas-brousse** [comment]
When I did a quick check yesterday on a demo project I didn't see anything. My project is really small so however.

**Comment by nicolas-brousse** [comment]
@joelhawksley `ViewComponent` doesn't use [`ActionView::Template#render`](https://github.com/rails/rails/blob/528b62e3869288b5f3ffbd1491f10bad00958c08/actionview/lib/action_view/template.rb#L179). It seems that only this method are instrumented.

[`ActionView::Template::Handlers::ERB#call`](https://github.com/rails/rails/blob/528b62e3869288b5f3ffbd1491f10bad00958c08/actionview/lib/action_view/template/handlers/erb.rb#L31) has no instrumentation.

**Comment by Spone** [comment]
Closed by #910, apparently.
